python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -267,3 +268,110 @@ def unique_and_compact_node_pairs(
         unique_nodes = list(unique_nodes.values())[0]
 
     return unique_nodes, compacted_node_pairs
+
+
+def compact_node_pairs(
+    node_pairs: Union[
+        CSCFormatBase,
+        Dict[str, CSCFormatBase],
+    ],
+    dst_nodes: Union[
+        torch.Tensor,
+        Dict[str, torch.Tensor],
+    ],
+):
+    """
+    Compact node pairs and return original_row_ids (per type).
+
+    Parameters
+    ----------
+    node_pairs : Union[CSCFormatBase,
+                    Dict(str, CSCFormatBase)]
+        Node pairs representing source-destination edges.
+        - If `node_pairs` is a CSCFormatBase: It means the graph is
+        homogeneous. Also, indptr and indice in it should be torch.tensor
+        representing source and destination pairs in csc format. And IDs inside
+        are homogeneous ids.
+        - If `node_pairs` is a Dict(str, CSCFormatBase): The keys
+        should be edge type and the values should be csc format node pairs.
+        And IDs inside are heterogeneous ids.
+    dst_nodes: torch.Tensor or Dict[str, torch.Tensor]
+        Nodes of all destination nodes in the node pairs.
+        - If `dst_nodes` is a tensor: It means the graph is homogeneous.
+        - If `dst_nodes` is a dictionary: The keys are node type and the
+        values are corresponding nodes. And IDs inside are heterogeneous ids.
+
+    Returns
+    -------
+    Tuple[original_row_node_ids, compacted_node_pairs]
```

Comment:
Let's avoid using node_pairs in this helper function such that in the future cleanup you don't need clean up too many code.